### PR TITLE
fix(input-date-picker): update the focus and hover chevron icon color

### PR DESCRIPTION
**Pull request by aPreciado88**
**Related Issue:** [7712](https://github.com/Esri/calcite-design-system/issues/7712)

### Summary
This updates `input-date-picker` chevron colors to be `--calcite-color-text-3` when idle and `--calcite-color-text-1` when hovered or pressed.

### CODE REVIEW

**Review by benelan — COMMENTED** [review]
First PR lets go! :tada: 

The PR title is a bit too long and it shouldn't end with any punctuation. Also, we want to keep implementation details out of the PR title. Maybe change it to something like:

> fix(input-date-picker): consistent chevron icon color

The implementation details (like the design tokens used) can go in the PR body. And the tokens should be `--calcite-color-text-N` like you have in the code, rather than the legacy `--calcite-ui-text-N` tokens that are currently in the PR title/body.

Once you are done making changes, add the `pr ready for visual snapshots` label to run Chromatic.

**Review by benelan — CHANGES_REQUESTED** [review]
I took a closer look at the issue, and it doesn't say anything about the `open` property effecting the chevron color. For example, the current implementation wouldn't work for the Figma design's last example:

image

Can you take another look? The `:focus` and/or `:active` pseudo-classes may do what we want.

File: packages/calcite-components/src/components/input-date-picker/input-date-picker.tsx [hunk 1]
```diff
@@ -681,6 +681,7 @@ export class InputDatePicker
       // we set tab index to -1 to prevent delegatesFocus from stealing focus before we can set it
       <span class={CSS.toggleIcon} tabIndex={-1}>
         <calcite-icon
+          class={open ? "chevron-icon-up" : "chevron-icon-down"}
```
[the remark this answers is not in the document]
Comment (aPreciado88) [reply packages/calcite-components/src/components/input-date-picker/input-date-picker.tsx]: I consolidated this css into a single class and added it to [resources.ts](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/input-date-picker/resources.ts).

**Review by benelan — APPROVED** [review]
Nice changes, code LGTM! cc @ashetland for snapshot review

**Comment by ashetland** [comment]
> Nice changes, code LGTM! cc @ashetland for snapshot review

I'm going to approve the screenshots as all the changes are expected, but the change to `text.1` is only happening while I hover on the icon. It should change onHover and onFocus of the entire input. See PR #8955.

**Comment by aPreciado88** [comment]
@ashetland I updated the component to change the chevron's color when the input is hovered.

**Review by benelan — CHANGES_REQUESTED** [review]
@ashetland mentioned text-1 should be applied on `:focus` too

**Comment by aPreciado88** [comment]
@ashetland I updated the component to apply text-1 on :hover, :active and :focus

**Review by benelan — APPROVED** [review]
LGTM!

**Comment by ashetland** [comment]
Heyo, I'm still only seeing it on hover.

**Comment by jcfranco** [comment]
@aPreciado88 Can you update the PR title (specifically, the [description part](https://kapeli.com/cheat_sheets/Conventional_Commits.docset/Contents/Resources/Documents/index#:~:text=Commit%20Message%20Structure)) to use the imperative, present tense per the [conventions](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md?rgh-link-date=2024-04-03T00%3A01%3A43Z#subject) we link to in our contributing doc? You can look at previous [commits on `main`](https://github.com/Esri/calcite-design-system/commits/main) as examples.

**Comment by aPreciado88** [comment]
@jcfranco I updated the title! Please let me know if that looks good.

**Comment by ashetland** [comment]
🤘🚀🚢🎉

**Pull request merged**